pkg/component/controller/etcd.go

```diff
@@ -211,15 +211,15 @@ func (e *Etcd) Reconcile() error {
 	return nil
 }
 
-func (e *Etcd) setupCerts() error {
+func (e *Etcd) setupCerts(ctx context.Context) error {
 	etcdCaCert := filepath.Join(e.K0sVars.EtcdCertDir, "ca.crt")
 	etcdCaCertKey := filepath.Join(e.K0sVars.EtcdCertDir, "ca.key")
 
 	if err := e.CertManager.EnsureCA("etcd/ca", "etcd-ca"); err != nil {
 		return fmt.Errorf("failed to create etcd ca: %w", err)
 	}
 
-	eg, _ := errgroup.WithContext(context.Background())
+	eg, _ := errgroup.WithContext(ctx)
```

Comment:
do we even need to have WithContext here, instead of regular errgroup? the returned contexted is underscored and none of the goroutines use it

is it better to use `var eg errgroup.Group` here?

Reply:
I haven't found any reason to keep `errgroup.WithContext` here. Changed to `var eg errgroup.Group`